extjsdk/README.md

```diff
@@ -25,11 +25,27 @@
+
+If the WebSocket connection closes through any means other than a call of `client.stop` then all Futures are forcibly completed as false, the program is reset to just before step 3 (maintaining any changes to handlers) and closeHandler is called. At this point, you should start step 3 again.
+
+### Details on initiateFullConnection
+The call `future = client.initiateFullConnection` is identical to `client.initiateWebSocketConnection; client.authenticate; future = client.connectToSource`. Each of these functions returns their own CompletableFuture, which is created whenever a call to it is made and either no previous call has been made or the previous call has already failed. These Futures can be obtained without the risk of (re)creating them through their own getters. Before the first call and after a Websocket connection closure each Future is null. <br>
+The Futures for these three functions are chained together. That is, if all three are called one of the three fails, then its Future completes as false and so do the futures of all the subsequent steps. Also, if a step is called before the previous connection step completes (but after the previous function has been called) then that step will automatically startup once the previous Future completes successfully.
+
+	
 ## <a name="client" id="client"></a>Using ExtensionWebSocketClient
 Every ExtensionWebSocketClient (Client) deals with a single source across its own WebSocket connection. 
 
 ### Connecting to a source
-Connection is as simple as creating a client with `new ExtensionWebSocketClient(<source name>)` and then calling `client.inititiateFullConnection(<vantiq url>, <authentication token>)`, where the authentication token is created by an admin using the Vantiq client. The call returns a CompletableFuture that will return whether or not the connection succeeded. If it succeeds, you can now send and receive messages from the source. If it fails, you can either try connecting again, or call `isOpen()`, `isAuthed()`, and `isConnected()` to see if the connection succeeded or failed at the WebSocket, authentication, and source levels respectively.
+Connection is as simple as creating a client with `new ExtensionWebSocketClient(<source name>)` and then calling `client.initiateFullConnection(<vantiq url>, <authentication token>)`, where the authentication token is created by an admin using the Vantiq client. The call returns a CompletableFuture that will return whether or not the connection succeeded. If it succeeds, you can now send and receive messages from the source. If it fails, you can either try connecting again, or call `isOpen()`, `isAuthed()`, and `isConnected()` to see if the connection succeeded or failed at the WebSocket, authentication, and source levels respectively.
```

Comment:
There was an authorize() call with user/pw.  Does that exist for initiateFullConnection() as well?  If not, it's worth mentioning somewhere that you may have to use the other mechanism.

Reply:
That's fine & I agree with your reasoning there.  But it should probably be mentioned somewhere in the documentation.  Thanks